sensor/common/scannerclient/grpc_client.go

```diff
@@ -143,25 +143,20 @@ func dialV2() (ScannerClient, error) {
 
 // dialV4 connect to scanner V4 gRPC and return a new ScannerClient.
 func dialV4() (ScannerClient, error) {
-	endpoint, err := getScannerEndpoint(env.ScannerV4GRPCEndpoint)
+	ctx := context.Background()
```

Comment:
Adding a timeout to the context may prevent the dial from blocking forever (which I believe is the default), applies to `dialV2` as well (didn't catch this in previous review). For consideration.

Reply:
Notice: This has stayed the same from the original implementation (pre scanner-v4).

Reply:
Thumbs up, resolved.